node/src/components/network/config.rs

```diff
@@ -0,0 +1,46 @@
+use std::net::SocketAddr;
+
+use datasize::DataSize;
+use serde::{Deserialize, Serialize};
+
+use crate::components::small_network;
+
+const DEFAULT_BIND_ADDRESS: &str = "0.0.0.0:22777";
+
+impl Default for Config {
+    fn default() -> Self {
+        Config {
+            bind_address: DEFAULT_BIND_ADDRESS.to_string(),
+            known_addresses: Vec::new(),
+            systemd_support: false,
+        }
+    }
+}
+
+/// Peer-to-peer network configuration.
+#[derive(DataSize, Debug, Clone, Deserialize, Serialize)]
+// Disallow unknown fields to ensure config files and command-line overrides contain valid keys.
+#[serde(deny_unknown_fields)]
+pub struct Config {
```

Comment:
As state above, we should redo the config.

Reply:
As above